src/sentry/lang/javascript/processor.py

```diff
@@ -697,26 +705,46 @@ def _lookup_in_open_archives(self, block):
 
         return None
 
-    def _get_debug_id_artifact_bundle_entry(self, debug_id, source_file_type):
+    def _get_artifact_bundle_entry_by_debug_id(self, debug_id, source_file_type):
         """
         Gets the DebugIdArtifactBundle entry that maps the debug_id and source_file_type to a specific ArtifactBundle.
 
         This query might return multiple entries, since a debug_id can point to multiple bundles. In this case we
         decided to take the first element from the result set, but we could use a stronger heuristic like taking the
         newest entry.
         """
-        # TODO: in the future we would like to load all the artifact bundles that are connected to the projects
-        #  we have permissions on not all the bundles.
-        return (
-            DebugIdArtifactBundle.objects.filter(
+        project_id = self.project.id if self.project else None
+
+        if (debug_id, source_file_type) in self.empty_result_for_debug_ids:
+            raise Exception(
+                f"There are no artifact bundles bound to project {project_id}"
+                f"that contain debug_id {debug_id} for source_file_type {source_file_type}"
+            )
+
+        entry = (
+            ArtifactBundle.objects.filter(
                 organization_id=self.organization.id,
-                debug_id=debug_id,
-                source_file_type=source_file_type.value if source_file_type is not None else None,
+                debugidartifactbundle__debug_id=debug_id,
+                debugidartifactbundle__source_file_type=source_file_type.value
+                if source_file_type is not None
+                else None,
+                projectartifactbundle__project_id=project_id,
             )
             .order_by("-date_added")
-            .select_related("artifact_bundle__file")[:1][0]
+            .select_related("file")[:1]
         )
 
+        # In case we didn't find any matching result, we will cache that this query had an empty result in order to
+        # avoid making the query for subsequent frames.
+        if len(entry) == 0:
+            self.empty_result_for_debug_ids.add((debug_id, source_file_type))
+            raise Exception(
+                f"There are no artifact bundles bound to project {self.project.id}"
+                f"that contain debug_id {debug_id} for source_file_type {source_file_type}"
+            )
+
+        return entry[0]
```

Comment:
You can use `.first()` in the query and then check for `if entry is None:` instead. Easier to read imo than using indexes in few places.

Reply:
We just found a small issue in the implementation and now we immediately compute the `list` out of the results, thus we use the `[]` indexing operator now.